server/application/application.go

```diff
@@ -1571,10 +1586,11 @@ func (s *Server) Rollback(ctx context.Context, rollbackReq *application.Applicat
 		},
 	}
 	a, err = argo.SetAppOperation(appIf, *rollbackReq.Name, &op)
-	if err == nil {
-		s.logAppEvent(a, ctx, argo.EventReasonOperationStarted, fmt.Sprintf("initiated rollback to %d", rollbackReq.GetId()))
+	if err != nil {
+		return a, fmt.Errorf("error setting app operation: %w", err)
```

Comment:
I think you can explicitly return nil, like in the refactor above. `SetAppOperation` won't return a non-nil value with an error.